src/simsopt/field/mgrid.py

```diff
@@ -0,0 +1,290 @@
+
+        # add fields
+        for j in np.arange(self.n_ext_cur):
+
+            tag = '_%.3i' % (j+1)
+            var_br_001 = ds.createVariable('br'+tag, 'f8', ('phi', 'zee', 'rad'))
+            var_bp_001 = ds.createVariable('bp'+tag, 'f8', ('phi', 'zee', 'rad'))
+            var_bz_001 = ds.createVariable('bz'+tag, 'f8', ('phi', 'zee', 'rad'))
+
+            var_br_001[:, :, :] = self.br_arr[j]
+            var_bz_001[:, :, :] = self.bz_arr[j]
+            var_bp_001[:, :, :] = self.bp_arr[j]
+
+        ds.close()
+
+    @classmethod 
+    def from_file(cls, filename):
+        '''
+        This method reads MGrid data from file.
+
+        Args:
+            filename: mgrid netCDF input file name
+        '''
+
+        f = netcdf_file(filename, 'r', mmap=False)
+
+        # load grid
+        nr = f.variables['ir'].getValue()
+        nphi = f.variables['kp'].getValue()
+        nz = f.variables['jz'].getValue()
+        rmin = f.variables['rmin'].getValue()
+        rmax = f.variables['rmax'].getValue()
+        zmin = f.variables['zmin'].getValue()
+        zmax = f.variables['zmax'].getValue()
+        kwargs = {"nr": nr, "nphi": nphi, "nz": nz, 
+                  "rmin": rmin, "rmax": rmax, "zmin": zmin, "zmax": zmax}
+
+        mgrid = cls(**kwargs)
+        mgrid.filename = filename
+        mgrid.n_ext_cur = int(f.variables['nextcur'].getValue())
+        coil_data = f.variables['coil_group'][:]
+        mgrid.coil_names = [_unpack(coil_data[j]) for j in range(mgrid.n_ext_cur)] 
+
+        mgrid.mode = f.variables['mgrid_mode'][:][0].decode()
+        mgrid.raw_coil_current = np.array(f.variables['raw_coil_cur'][:])
+
+        br_arr = []
+        bp_arr = []
+        bz_arr = []
+
+        nextcur = mgrid.n_ext_cur
+        for j in range(nextcur):
+            idx = '{:03d}'.format(j+1)
+            br = f.variables['br_'+idx][:]  # phi z r
+            bp = f.variables['bp_'+idx][:]
+            bz = f.variables['bz_'+idx][:]
+
+            if (mgrid.mode == 'S'):
+                br_arr.append(br * mgrid.raw_coil_current[j])
+                bp_arr.append(bp * mgrid.raw_coil_current[j])
+                bz_arr.append(bz * mgrid.raw_coil_current[j])
+            else:
+                br_arr.append(br)
+                bp_arr.append(bp)
+                bz_arr.append(bz)
+
+        mgrid.br_arr = np.array(br_arr)
+        mgrid.bp_arr = np.array(bp_arr)
+        mgrid.bz_arr = np.array(bz_arr)
+
+        # sum over coil groups
+        if nextcur > 1:
+            br = np.sum(br_arr, axis=0)
+            bp = np.sum(bp_arr, axis=0)
+            bz = np.sum(bz_arr, axis=0)
+        else:
+            br = br_arr[0]
+            bp = bp_arr[0]
+            bz = bz_arr[0]
+
+        mgrid.br = br
+        mgrid.bp = bp
+        mgrid.bz = bz
+
+        mgrid.bvec = np.transpose([br, bp, bz])
+
+        return mgrid
+
+    def plot(self, jphi=0, bscale=0, show=True):
+        '''
```

Comment:
If `show=False`, nothing useful happens. It should return a plot that could be saved.

Reply:
Done.